tests/views/test_instructor_file.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the `fonzie` instructor_file module.
+"""
+# pylint: disable=no-member,import-error
+from __future__ import absolute_import, unicode_literals
+
+import hashlib
+
+import six
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from django.core.urlresolvers import reverse
+
+from student.roles import CourseStaffRole
+from student.tests.factories import UserFactory
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class InstructorFilesViewTestCase(ModuleStoreTestCase, APITestCase):
+    """Tests for the InstructorFilesView API endpoint"""
+
+    def setUp(self):
+        """
+        Set common parameters for unit tests.
+
+        """
+        super(InstructorFilesViewTestCase, self).setUp()
+
+        self.filename = "fun_test_00_student_profile_info_2019-04-19-1406.csv"
+        self.course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+        self.course_staff = UserFactory(
+            username="course_staff", email="course_staff@fun-mooc.fr"
+        )
+        CourseStaffRole(self.course.id).add_users(self.course_staff)
+        self.course_sha1 = hashlib.sha1(six.text_type(self.course.id)).hexdigest()
+
+        self.url = reverse(
+            "fonzie:protected:instructor_file",
+            kwargs={
+                "version": "1.0",
+                "course_sha1": self.course_sha1,
+                "filename": self.filename,
+            },
+        )
+
+    def test_user_is_not_logged_in(self):
+        """
+        User should be logged in, if not the view should return 403
+        """
+        response = self.client.get(self.url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_user_is_not_staff(self):
+        """
+        User should not be a just student, if so the view should return 403
+        """
+
+        normal_user = UserFactory(
+            username="normal_user", email="normal_user@fun-mooc.fr"
+        )
+        self.client.force_authenticate(normal_user)
+        response = self.client.get(self.url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_user_is_staff_on_other_course(self):
+        """
+        User should belong to this course staff, if not the view should return 403
+        """
+
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+        other_course_staff = UserFactory(
+            username="other_course_staff", email="other_course_staff@fun-mooc.fr"
+        )
+        CourseStaffRole(course.id).add_users(other_course_staff)
```

Comment:
I have no idea what this code does. Hence it's hard to know if it works or if it is a duplicate of the previous test. Maybe add a comment?